src/parser/mod.rs

```diff
@@ -7928,6 +7930,24 @@ impl<'a> Parser<'a> {
                 vec![]
             };
 
+            let settings = if dialect_of!(self is ClickHouseDialect|GenericDialect)
+                && self.parse_keyword(Keyword::SETTINGS)
+            {
+                let mut key_values: Vec<Setting> = vec![];
+                loop {
+                    let key = self.parse_identifier(false)?;
+                    self.expect_token(&Token::Eq)?;
+                    let value = self.parse_value()?;
+                    key_values.push(Setting { key, value });
+                    if !self.consume_token(&Token::Comma) {
+                        break;
+                    }
+                }
+                Some(key_values)
```

Comment:
I think we might be able to reuse the [parse_comma_separated](https://github.com/sqlparser-rs/sqlparser-rs/blob/cdba898cfe6e0aed073655f0d3a56e38eed4eb75/src/parser/mod.rs#L3338) functionality here, could you take a look at that function to see if it works for this case?

Reply:
@iffyio Good point, it's done now.